exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/PlannerSettings.java

```diff
@@ -133,6 +133,9 @@
      the need to turn off join optimization may go away.
    */
   public static final BooleanValidator JOIN_OPTIMIZATION = new BooleanValidator("planner.enable_join_optimization", true);
+  // for testing purpose
```

Comment:
@VisibleForTesting annotates methods; but this is a session option. 
Also (hidden) is the possibility that this option may be used in production in case some query yields a single phase hashagg but still has too much data to handle.